### PR TITLE
User Orders Index

### LINKED ISSUE #55: User Story 28, User Profile displays Orders
```
As a registered user
When I visit my Profile Orders page, "/profile/orders"
I see every order I've made, which includes the following information:
- the ID of the order, which is a link to the order show page
- the date the order was made
- the date the order was last updated
- the current status of the order
- the total quantity of items in the order
- the grand total of all items for that order
```

DO THESE STORIES FIRST:
- User Story 17, User Profile Show Page

### Checklist:

- [x] base tests are written
- [x] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #241

child of #274

depends on #252

### PR DESCRIPTION
**Pull request by BrennanAyers**
This PR brings in the functionality of a Default User being able to see all of their Orders collated into one page. These sections will contain the Order ID, which will link to an Order show page, the Date Made, the Last Update date, the Current Status, the Number of Items, and the Grand Total of the order.
This required making new instance methods for an Order for #date_made, #last_updated, #item_count, and #grand_total. Most of these just access the Order or its Items directly, but the Grand Total required accessing the Order Items and performing a Sum on them based on quantity and price. We have also implemented an Enum for the Order Status, for Pending, Packaged, Shipped, and Cancelled.
The HTML for the page is currently simple, but can easily be converted to Card format when the time comes.

Resolves #55

### CODE REVIEW

**Comment by BrennanAyers** [comment]
Specs are all passing after Merge from Master.

File: config/routes.rb [hunk 1]
```diff
@@ -15,11 +15,10 @@
...
   get '/profile', to: 'users#show'
   get '/profile/edit', to: 'users#edit'
-  patch 'profile/edit', to: 'users#update'
+  patch '/profile/edit', to: 'users#update'
+  get '/profile/orders', to: 'orders#index'
 
-  resources :users, as: :profile do  #add for User Story #27, User Profile displays Orders link
-    resources :orders, only: [:index]
-  end
+  resources :orders, only: [:show]
```
Comment (BrennanAyers) [inline config/routes.rb:21]: @WHomer Redid this route because it didn't follow our naming convention.

File: spec/features/users/show_spec.rb [hunk 2]
```diff
@@ -44,7 +44,7 @@
 
           click_on "My Orders"
 
-          expect(current_path).to eq(profile_orders_path(@user))
+          expect(current_path).to eq(profile_orders_path)
```
Comment (BrennanAyers) [inline spec/features/users/show_spec.rb:47]: Also redoing the Route allowed for us to not have to pass in the user, as it uses current_user. Should be fine for now, we'll see how it works with Admin @WHomer

File: spec/models/order_spec.rb [hunk 3]
```diff
@@ -9,4 +11,75 @@
...
+
+    it 'last_updated' do
+      travel_to Time.zone.local(2019, 04, 12, 8, 00, 00)
+      @order_1.update(status: 2)
+      travel_to Time.zone.local(2019, 05, 11, 18, 00, 00)
+      @order_2.update(status: 2)
+      travel_to Time.zone.local(2019, 05, 03, 12, 00, 00)
+      @order_3.update(status: 2)
+      travel_to Time.zone.local(2018, 01, 16, 14, 00, 00)
+      @order_4.update(status: 2)
```
Comment (VinceCarollo) [inline spec/models/order_spec.rb]: Interesting way to create orders in a time zone. Another way would be to just set the created_at attribute.

File: app/models/order.rb [hunk 4]
```diff
@@ -4,4 +4,24 @@ class Order < ApplicationRecord
...
+
+  enum status: [:pending, :packaged, :shipped, :cancelled]
+
+  def date_made
+    created_at.strftime("%B %-d, %Y")
+  end
+
+  def last_updated
+    updated_at.strftime("%B %-d, %Y")
+  end
```
Comment (VinceCarollo) [inline app/models/order.rb:16]: This will be useful.

**Review by VinceCarollo — APPROVED** [review]
